src/Umbraco.Core/IO/ShadowFileSystemsScope.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.Remoting.Messaging;
+using Umbraco.Core.Logging;
+
+namespace Umbraco.Core.IO
+{
+    public class ShadowFileSystemsScope : IDisposable
```

Comment:
Is public so Deploy can begin shadowing the filesystems. Cannot do it "simply" by getting an `IDisposable` thing because we need to be able to "complete" the scope before disposing. So... the _class_ could be made internal but we'd still need to expose an interface of some sort. Which could be filesystem-independent, really it's something like:
~~~~
public interface ICompletable : IDisposable
{
    void Complete();
}
~~~~
Would that be better?